internal/csi-addons/rbd/replication.go

```diff
@@ -649,6 +652,28 @@ func (rs *ReplicationServer) ResyncVolume(ctx context.Context,
 	return resp, nil
 }
 
+func getError(err error) error {
+	errorStatusMap := map[error]codes.Code{
+		corerbd.ErrFetchingLocalState:          codes.Internal,
+		corerbd.ErrResyncImageFailed:           codes.Internal,
+		corerbd.ErrDisableImageMirroringFailed: codes.Internal,
+		corerbd.ErrFetchingMirroringInfo:       codes.Internal,
+		corerbd.ErrInvalidArgument:             codes.InvalidArgument,
+		corerbd.ErrAborted:                     codes.Aborted,
+		corerbd.ErrFailedPrecondition:          codes.FailedPrecondition,
+		corerbd.ErrUnavailable:                 codes.Unavailable,
+	}
+
+	for e, code := range errorStatusMap {
+		if errors.Is(err, e) {
+			return status.Error(code, err.Error())
+		}
+	}
+
+	// Handle any other error not listed in the map
+	return status.Error(codes.Unknown, err.Error())
```

Comment:
This is panic if the passed err is nil?

Reply:
No, this is a kind of defualt statement in switch. If the passed err is not among the ones listed in map and is not nil.

Reply:
can you add a unit test for a nil error?

Reply:
We are calling this function only if the passed error is not nil, that's why we don't have any handling for nil here.

Reply:
Adding a unit test for nil, it will send the unknown error otherwise.

Reply:
Please add it as a documentation to the function and also mention this function will panic if you pass the non-nil error

Reply:
If `err` is `nil`, the this should probably return a gRPC `code.OK` with `code.OK.String()` as message.

Reply:
Updated the function with nil check and added the same case in unit test.